FXIOS-1352 ⁃ FXIOS-386 - fixes #6498 Add report site issue option

chriscoderdr opened the pull request:
*Thank you for submitting a pull request, your contributions are greatly appreciated!*

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Please make sure you've run the test scheme and all the tests pass. If your changes affect existing tests please make sure to update the tests as well.
fixes

Fixes #6498  Add report site issue option, sending app version and os version too. Added the URL construction in SupportUtils.

Example report url: https://webcompat.com/issues/new?src=mobile-reporter&browser=Fennec%20(chrisgomez)-0.0.1&os=iOS-14.1.0&url=https%3A//www.bbc.com/news/uk-55475240

![simulator_screenshot_1ABC7B7A-360A-40A8-A1F9-DEE888BC14FE](https://user-images.githubusercontent.com/2073238/103301126-a0839280-49ce-11eb-9934-cf73992559c4.png)

![simulator_screenshot_7CE1CE26-E39A-4647-B51A-1B9DA16EE07D](https://user-images.githubusercontent.com/2073238/103301134-a7aaa080-49ce-11eb-8236-82e64cdb2a78.png)

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1352)

dnarcese commented on Client/Frontend/Widgets/PhotonActionSheet/PageActionMenu.swift:
```diff
@@ -209,13 +210,21 @@ extension PhotonActionSheetProtocol {
+            
+            commonActions.insert(reportSiteIssueAction, at: 0)
         }
+        
```
nit: remove empty line

chriscoderdr replied:
Removed

dnarcese commented on Shared/SupportUtils.swift:
```diff
@@ -18,4 +18,20 @@ public struct SupportUtils {
+              let browser = "\(AppInfo.displayName)-\(AppInfo.appVersion)".addingPercentEncoding(withAllowedCharacters: .urlPathAllowed)
+        else {
+            return nil
+            
```
nit: remove empty line

chriscoderdr replied:
Removed

dnarcese reviewed (commented):
What does the browser and os information provide to webcompat? I didn't notice it being used in the report.

chriscoderdr commented:
> What does the browser and os information provide to webcompat? I didn't notice it being used in the report.

Yes you're right I misunderstood the documentation, only two parameters working right now are URL and src
as per https://github.com/webcompat/webcompat.com/issues/2525#issuecomment-403676858
made the change now it's like:
https://webcompat.com/issues/new?src=mobile-reporter&url=https%3A//davy.dev/

dnarcese reviewed (commented):
I just noticed the colour of the icon is off. Its grey instead of black. Also, can you verify that the size is correct? Looks smaller than the others to me, but I didn't verify.

chriscoderdr commented:
> I just noticed the colour of the icon is off. Its grey instead of black. Also, can you verify that the size is correct? Looks smaller than the others to me, but I didn't verify.

Changed the fill color to black as the other icons in the menu
[link to the icon in the commit](https://github.com/mozilla-mobile/firefox-ios/blob/6095079ba4712d09334d4a6c3f990bcce58236b2/Client/Frontend/Menu/Menu.xcassets/menu-reportSiteIssue.imageset/Report%20Site%20-%20Light.pdf).
Also changed the image to remove the blank spaces around the icon the same as the other icons.

![Simulator Screen Shot - iPhone 11 - 2021-01-07 at 23 51 06](https://user-images.githubusercontent.com/2073238/103976036-b2d69c80-5143-11eb-8baa-ffc6442e126c.png)

gaganmastana13 commented:
On Thu, 7 Jan 2021 at 8:56 PM Cristian Gomez <notifications@github.com>
wrote:

> I just noticed the colour of the icon is off. Its grey instead of black.
> Also, can you verify that the size is correct? Looks smaller than the
> others to me, but I didn't verify.
>
> Changed the fill color to black as the other icons in the menu
> link to the icon in the committee
> <https://github.com/mozilla-mobile/firefox-ios/blob/6095079ba4712d09334d4a6c3f990bcce58236b2/Client/Frontend/Menu/Menu.xcassets/menu-reportSiteIssue.imageset/Report%20Site%20-%20Light.pdf>
>  .
> Also changed the image to remove the blank spaces around the icon the same
> as the other icons.
>
> [image: Simulator Screen Shot - iPhone 11 - 2021-01-07 at 23 51 06]
> <https://user-images.githubusercontent.com/2073238/103976036-b2d69c80-5143-11eb-8baa-ffc6442e126c.png>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla-mobile/firefox-ios/pull/7796#issuecomment-756545835>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASAPY3M2SPONWLXAUYCJ6UDSY2F7RANCNFSM4VNNIYKA>
> .
>
-- 
gaganmastana

dnarcese reviewed (commented):
I think there is an issue with image asset. The strokes look thicker than the rest of the icons. I am going to bring this up with our designer to see if it can be fixed. I will leave this PR open until then.

dnarcese commented:
@brampitoyo The lightbulb icon has slightly thicker outline than the rest of the icons. Let me know if there is a different asset we can use or what we should do next. Thanks

gaganmastana13 commented:
Download the new Google Pay app with my invite link - we’ll both earn
rewards when you make your first qualifying payment. While supplies last!.
https://g.co/payinvite/co3px13

brampitoyo commented:
@dnarcese The asset file I have for “Report site” are these:

- SVG: https://drive.google.com/file/d/1b6OleI6JzZplRXwAk3_y0SwEAB7IjA7q/view?usp=sharing
- PDF: https://drive.google.com/file/d/1b6P6nDutLJrcn3vZw01ikI36JSpFIuMo/view?usp=sharing

My assets look identical to the asset posted above, but it’s square-sized and contains additional empty space as borders. This will help make this icons look identical to the rest of our icons.

dnarcese commented:
@brampitoyo This is what the new icon looks like. It looks smaller than the others now. Are you able to fix that?
![IMG_05DB530C4E33-1](https://user-images.githubusercontent.com/11432165/108765065-5b3c9680-7521-11eb-80ab-69d5b7da7b94.jpeg)

brampitoyo reviewed (approved):
@dnarcese I can confirm that the original Report Site asset looks smaller than some icons, but compared to all other icons, it does look balanced. We should keep using the one supplied. Thanks!
